### PR TITLE
Feature/unity 1372 physical hands UI sliders

### PR DESCRIPTION
**Pull request by BryceNewell-UL**
## Summary

_Add a Summary here._

## Contributor Tasks

- [x] Create or edit test cases [here](https://ultrahaptics.atlassian.net/projects/UNITY?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testPlayer/UNITY-R147)
- [x] Add a CHANGELOG entry for this change.
- [x] Ensure documentation requirements are met e.g., public API is commented.
- [x] Consider any licensing/other legal implications for this MR e.g. notices required by any new libraries.

## Test Cycle

_Link to the test cycle here._

## Reviewer Tasks

- [x] Code reviewed.
- [x] Non-code assets e.g. Unity assets/scenes reviewed.
- [x] All tests must be ran and cover all scenarios (If not, add new tests to the cycle and run them).
- [ ] Documentation has been reviewed.
- [ ] Approve with a comment of any additional tests run or any observations.

## Related JIRA Issues

_If this MR closes any JIRA issues list them below in the form `Closes PROJECT-#`_

## Pull Request Templates

Switch template by going to preview and clicking the link - note it will not work if you've made any changes to the description.

- [default.md](?expand=1) - for contributions to stable packages.
- [release.md](?expand=1&template=release.md) - for release merge requests.

**You are currently using: default.md**

Note: these links work by overwriting query parameters of the current url. If the current url contains any you may want to amend the url with `&template=name.md` instead of using the link. See [query parameter docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/using-query-parameters-to-create-a-pull-request) for more information.

### CODE REVIEW

File: Packages/Tracking/Physical Hands/Runtime/Scripts/UI/PhysicalHandsUISlider.cs [hunk 1]
```diff
@@ -0,0 +1,835 @@
...
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Leap.Unity.PhysicalHands
+{
+    [RequireComponent(typeof(Rigidbody))]
+    public class PhysicalHandsUISlider : MonoBehaviour
```
Comment (MattGrayUL) [inline Packages/Tracking/Physical Hands/Runtime/Scripts/UI/PhysicalHandsUISlider.cs:12]: We can probably remove UI from the name here

File: Packages/Tracking/Physical Hands/Runtime/Scripts/UI/PhysicalHandsUISlider.cs [hunk 2]
```diff
@@ -0,0 +1,835 @@
...
+                                break;
+                            }
+                    }
+
+                    // Reset flag and update previous slider value
+                    _sliderReleasedLastFrame = false;
+                    _prevSliderValue = _sliderValue;
+                }
+                // Send slider change event
+                SendSliderEvent(SliderChangeEvent, TwoDimSliderButtonPressedEvent);
```
Comment (MattGrayUL) [inline Packages/Tracking/Physical Hands/Runtime/Scripts/UI/PhysicalHandsUISlider.cs:233]: what about 1d sliders?

File: Packages/Tracking/Physical Hands/Runtime/Scripts/UI/PhysicalHandsUISlider.cs [hunk 3]
```diff
@@ -0,0 +1,835 @@
...
+            if (_freezeIfNotActive == false)
+            {
+                UnFreezeSliderPosition();
+            }
+            else
+            {
+                FreezeSliderPosition();
+            }
+
+            UpdateSliderPos(_startPosition, _twoDimStartPosition);
```
Comment (MattGrayUL) [inline Packages/Tracking/Physical Hands/Runtime/Scripts/UI/PhysicalHandsUISlider.cs:383]: what about 1d sliders?

File: Packages/Tracking/Physical Hands/Runtime/Scripts/UI/PhysicalHandsUISlider.cs [hunk 4]
```diff
@@ -0,0 +1,835 @@
...
+                    if (_twoDimNumberOfSegments.y != 0)
+                    {
+                        closestStep = GetClosestStep(_sliderValue.z, (int)_twoDimNumberOfSegments.y);
+                        twoDimValue.y = _sliderZZeroPos + (closestStep * (localSliderTravelDistance * 2));
+                    }
+                    break;
+            }
+
+            // Update the slider position
+            UpdateSliderPos(closestStep, twoDimValue);
```
Comment (MattGrayUL) [inline Packages/Tracking/Physical Hands/Runtime/Scripts/UI/PhysicalHandsUISlider.cs:808]: what about 1d sliders?

File: Packages/Tracking/Physical Hands/Runtime/Scripts/UI/PhysicalHandsUISlider.cs (hunk 2 above)
Comment (BryceNewell-UL) [reply Packages/Tracking/Physical Hands/Runtime/Scripts/UI/PhysicalHandsUISlider.cs:233]: was actually the 2d sliders broken here, meant to be sending a changed event, send slider event sends either event depending on the type of slider

File: Packages/Tracking/Physical Hands/Runtime/Scripts/UI/PhysicalHandsSlider.cs [hunk 5]
```diff
@@ -0,0 +1,642 @@
...
+                case SliderDirection.Z:
+                _configurableJoint.zMotion = ConfigurableJointMotion.Limited;
+                    break;
+            }
+        }
+
+
+        #endregion
+
+        private void OnDrawGizmosSelected()
```
Comment (MattGrayUL) [inline Packages/Tracking/Physical Hands/Runtime/Scripts/UI/PhysicalHandsSlider.cs:386]: move gizmos to the bottom of the file?

**Review by MattGrayUL — APPROVED** [review]

**Pull request merged**